databags/databag_timer.cpp

```diff
@@ -0,0 +1,55 @@
+#include "databag_timer.h"
+
+void TimersDatabag::_bind_methods() {
+}
+
+void TimersDatabag::set_now(const uint64_t newNow){
+	now = newNow;
+}
+
+void TimersDatabag::internal_restart_precise_timer(const godex::TimerHandle timer, const uint64_t microseconds) {
+	timers[timer] = microseconds;
+}
+
+godex::TimerHandle TimersDatabag::new_timer(const real_t seconds) {
+	return new_precise_timer((uint64_t)((double)seconds * 1000000.0));
+}
+
+godex::TimerHandle TimersDatabag::new_precise_timer(const uint64_t microseconds) {
+	if (destroyed_timers.size() > 0) {
+		TimerHandle finalHandle = destroyed_timers[destroyed_timers.size() - 1];
+		internal_restart_precise_timer(finalHandle, microseconds);
+		destroyed_timers.remove_unordered(destroyed_timers.size() - 1);
+		return finalHandle;
+	} else {
+		timers.push_back(now + microseconds);
+		return godex::TimerHandle(timers.size() - 1);
+	}
+}
+
+void TimersDatabag::restart_timer(const godex::TimerHandle timer, const real_t seconds) {
+	restart_precise_timer(timer, (uint64_t)((double)seconds * 1000000.0));
+}
+
+void TimersDatabag::restart_precise_timer(const godex::TimerHandle timer, const uint64_t microseconds) {
+	ERR_FAIL_COND(destroyed_timers.find(timer) != -1);
+	internal_restart_precise_timer(timer, microseconds);
+}
+
+bool TimersDatabag::is_done(const godex::TimerHandle timer) const {
+	return timers[timer] < now;
```

Comment:
Add an `ERR_FAIL_COND_V_MSG` to make sure the `p_timer` is a valid handle.

Reply:
I think local_vector already has an errfailcondition on [] access? Do we want to check twice?

Reply:
```
	_FORCE_INLINE_ const T &operator[](U p_index) const {
		CRASH_BAD_UNSIGNED_INDEX(p_index, count);
		return data[p_index];
	}
```
It crashes if you do so, better to guard it with an error there.